### PR TITLE
Migration/keysignature

### PR DESCRIPTION
**Pull request by rvilarl**
keysignature is now migrated, however the result is different, the signatures are now more compact.
@0xfe @ronyeh please look if I introduced a bug.

The reason might also be a bug in the code `KeySigNote` was accessing `width` directly in `keysignature` and not through `getWidth`. As a consequence, `format` was not called.

### CODE REVIEW

**Comment by rvilarl** [comment]
**current**

Clef_Keys Major_Key_Clef_Test_Current

**reference**

Clef_Keys Major_Key_Clef_Test_Reference

**Comment by rvilarl** [comment]
Conflicts resolved

**Comment by rvilarl** [comment]
> keysignature is now migrated, however the result is different, the signatures are now more compact.
> @0xfe @ronyeh please look if I introduced a bug.
> 
> The reason might also be a bug in the code `KeySigNote` was accessing `width` directly in `keysignature` and not through `getWidth`. As a consequence, `format` was not called.

My guess was not correct, the difference has nothing to do with `getWidth` and `format`
@ronyeh could you make a review?

**Comment by rvilarl** [comment]
@0xfe @ronyeh please look into that, I really do not see the difference.

**Comment by rvilarl** [comment]
@0xfe this is the last class to get the src migration complete!

File: src/keysignature.ts [hunk 1]
```diff
@@ -114,11 +124,11 @@ export class KeySignature extends StaveModifier {
     }
 
     // Place the glyph on the stave
-    this.placeGlyphOnLine(glyph, this.stave, acc.line);
+    this.placeGlyphOnLine(glyph, this.checkStave(), acc.line);
     this.glyphs.push(glyph);
 
     const xPosition = this.xPositions[this.xPositions.length - 1];
-    const glyphWidth = glyph.getMetrics().width + extraWidth;
+    const glyphWidth = glyph.getMetrics().width ?? 0 + extraWidth;
```
Comment (0xfe) [inline src/keysignature.ts:131]: Can we remove the `?? 0` -- there could be a silent failure here.

File: src/keysignote.ts [hunk 2]
```diff
@@ -25,7 +25,7 @@ export class KeySigNote extends Note {
     this.setPreFormatted(true);
     this.keySignature.setStave(this.checkStave());
     this.keySignature.format();
```
Comment (0xfe) [inline src/keysignote.ts:27]: You can remove this (since `getWidth` calls `format`).

File: src/keysignature.ts (hunk 1 above)
Comment (rvilarl) [reply src/keysignature.ts:131]: done

File: src/keysignote.ts (hunk 2 above)
Comment (rvilarl) [reply src/keysignote.ts]: done

**Comment by rvilarl** [comment]
@0xfe ready

**Comment by 0xfe** [comment]
Does the same visual diff show up?

**Comment by rvilarl** [comment]
@0xfe I believe so I will check

**Comment by rvilarl** [comment]
@0xfe No! There is no difference, you fixed it! Welldone!

**Comment by 0xfe** [comment]
👍 great!

**Pull request merged**